components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2444,12 +2446,13 @@ public void createNewAPIVersion(API api, String newVersion) throws DuplicateAPIE
             String resourcePath = APIUtil.getOpenAPIDefinitionFilePath(api.getId().getApiName(),
                     api.getId().getVersion(), api.getId().getProviderName());
             if (registry.resourceExists(resourcePath + APIConstants.API_OAS_DEFINITION_RESOURCE_NAME)) {
-                JSONObject swaggerObject = (JSONObject) new JSONParser()
-                        .parse(definitionFromOpenAPISpec.getAPIDefinition(api.getId(), registry));
-                JSONObject infoObject = (JSONObject) swaggerObject.get("info");
+                String apiDefinition = definitionFromOpenAPISpec.getAPIDefinition(api.getId(), registry);
+                LinkedHashMap map = new ObjectMapper().readValue(apiDefinition, LinkedHashMap.class);
```

Comment:
Descriptive name for variable map :)

Reply:
Fixed with 32e0ffac6b929366b9b43de63779401cc71b04ec